packages/@lwc/babel-plugin-component/src/component.ts

```diff
@@ -45,7 +45,8 @@ function needsComponentRegistration(path: DeclarationPath) {
 
 function getComponentRegisteredName(t: BabelTypes, state: LwcBabelPluginPass) {
     const { namespace, name } = state.opts;
-    const componentName = namespace && name ? `${namespace}-${name}` : '';
+    const kebabCasedName = name?.replace(/([a-z])([A-Z])/g, '$1-$2').toLowerCase();
```

Comment:
This is fine for now, but I wish we had a single module in `@lwc/shared` to handle camel-to-kebab and kebab-to-camel conversions, to ensure we do it consistently everywhere. Can we do this in a follow-up PR?

E.g. some other usages I found:

https://github.com/salesforce/lwc/blob/3770a62e261db4cafa6b82bfa10de21068e1395d/packages/%40lwc/template-compiler/src/shared/naming.ts#L7

https://github.com/salesforce/lwc/blob/3770a62e261db4cafa6b82bfa10de21068e1395d/packages/%40lwc/shared/src/html-attributes.ts#L189

Reply:
Yup!  I can create a follow-up PR to do this.